File: src/python/problem16.py
#!/usr/bin/env python


def pow(m,n):
	res=1
	while(n>0):
		res=res*m
		n-=1
	return str(res)	

res=pow(2,1000)

res=str(reduce((lambda x,y:x*y),[2 for i in range(0,1000)]))
print sum(int(res[i]) for i in range(0,len(res)))

File: src/java/Problem16.java
public class Problem16 {
	
	private static String mul(String s,int n) {
		String res="";
		int c=0;
		for(int i=0;i<s.length();i++) {
			int sum=((s.charAt(s.length()-i-1)-'0')*n)+c;
			res=(sum%10)+res;
			c=sum/10;
			
		}
		if(c>0) {
			res=c+res;
		}	
		return res;

	}

	public static String pow(int n,int m) {
		String s="1";
		while(m-->0) {
			s=mul(s,n);
		}
		return s;
	}
	public static int sum(String s) {
		int res=0;
		for(int i=0;i<s.length();i++) {
			res+=(s.charAt(i)-'0');
		}
		return res;
	}

	public static void main(String[] args) {
		System.out.println(sum(pow(2,1000)));
	}

}
